junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/LauncherDiscoveryResult.java

```diff
@@ -33,11 +35,13 @@
 
 	private final Map<TestEngine, TestDescriptor> testEngineDescriptors;
 	private final ConfigurationParameters configurationParameters;
+	private final Optional<TestDescriptor> parentDescriptor;
```

Comment:
Optional should not be used as a wrapper for fields. Consider pass in a null value in the constructor value and wrap it in the getter. As al alternative to passing an null value consider creating multiple constructors.